File: pkg/apiclient/library.go
// Package apiclient provides library API calls for the gtmpc REST API.
package apiclient

import "fmt"

// GetTracks fetches the full track library via GET /api/library/tracks.
// Requires authentication (token must be set via SetToken).
func (c *APIClient) GetTracks() (*TrackListResponse, error) {
	var resp TrackListResponse
	if err := c.doJSONRequest("GET", "/api/library/tracks", nil, &resp); err != nil {
		return nil, fmt.Errorf("get tracks: %w", err)
	}
	return &resp, nil
}

// GetPlaylists fetches all playlists via GET /api/library/playlists.
// Requires authentication.
func (c *APIClient) GetPlaylists() (*PlaylistListResponse, error) {
	var resp PlaylistListResponse
	if err := c.doJSONRequest("GET", "/api/library/playlists", nil, &resp); err != nil {
		return nil, fmt.Errorf("get playlists: %w", err)
	}
	return &resp, nil
}

// CreatePlaylist creates a new playlist via POST /api/library/playlists.
// Requires authentication.
func (c *APIClient) CreatePlaylist(req CreatePlaylistRequest) (*CreatePlaylistResponse, error) {
	var resp CreatePlaylistResponse
	if err := c.doJSONRequest("POST", "/api/library/playlists", req, &resp); err != nil {
		return nil, fmt.Errorf("create playlist: %w", err)
	}
	return &resp, nil
}

// CoverURL returns the URL for fetching cover art for a given track ID.
// Returns an empty string if trackID is empty.
func (c *APIClient) CoverURL(trackID string) string {
	if trackID == "" {
		return ""
	}
	return c.BaseURL + "/api/library/cover/" + trackID
}

File: web/src/api/library.ts
/**
 * src/api/library.ts
 * Library API calls: tracks, playlists, cover art, stream URL.
 */

import apiClient from './client';
import type {
  TrackListResponse,
  PlaylistListResponse,
  CreatePlaylistRequest,
  CreatePlaylistResponse,
} from './types';

export async function getTracks(): Promise<TrackListResponse> {
  const { data } = await apiClient.get<TrackListResponse>('/api/library/tracks');
  return data;
}

export async function getPlaylists(): Promise<PlaylistListResponse> {
  const { data } = await apiClient.get<PlaylistListResponse>('/api/library/playlists');
  return data;
}

export async function createPlaylist(req: CreatePlaylistRequest): Promise<CreatePlaylistResponse> {
  const { data } = await apiClient.post<CreatePlaylistResponse>('/api/library/playlists', req);
  return data;
}

/** Returns the URL to stream audio for a track. Used as <audio src>. */
export function streamURL(trackId: string): string {
  return `/api/stream/${trackId}`;
}

/** Returns the URL to fetch cover art for a track. */
export function coverURL(trackId: string): string {
  return `/api/library/cover/${trackId}`;
}
